custom_components/opnsense/update.py

```
@@ -263,71 +204,79 @@
         """Install an update."""
         state = self.coordinator.data
         upgrade_type = dict_get(state, "firmware_update_info.status")
         if upgrade_type not in ["update", "upgrade"]:
             return
 
-        client = self._get_opnsense_client()
-        task_details = client.upgrade_firmware(upgrade_type)
         sleep_time = 10
         running = True
         while running:
             time.sleep(sleep_time)
```

Ruff

Async functions should not call `time.sleep`

(ASYNC251)